### ISSUE TITLE
Same intervals for BAIT_INTERVALS and TARGET_INTERVALS giving unexpected different BAIT/TARGET ratio result

### ISSUE THREAD

**Issue by GATKSupportTeam** [broadinstitute/picard#1494]
This request was created from a contribution made by Ramzi TEMANNI on April 03, 2020 19:31 UTC.

Link: https://gatk.broadinstitute.org/hc/en-us/community/posts/360060681891-CollectHsMetrics-Exome-Data-

--

Can you please provide  
a) GATK version used  
b) Exact GATK commands used  
c) The entire error log if applicable.

When selecting the same intervals for BAIT\_INTERVALS and TARGET\_INTERVALS, I'm seeing different results where you would expect the same results BAIT/TARGET

There's a ratio of **1.98** between BAIT results and TARGET results

|
 BAIT\_TERRITORY 
 | 60456963 
 |   
 |
|
 TARGET\_TERRITORY 
 | 60456963 
 |   
 |
|
 PF\_BASES\_ALIGNED 
 | 6004024994 
 |   
 |
|
 PF\_UQ\_BASES\_ALIGNED 
 | 4441806737 
 |   
 |
|
   
 |   
 |   
 |
|
 ON\_BAIT\_BASES 
 | 3982752493 
 | **1.98478902** 
 |
|
 ON\_TARGET\_BASES 
 | 2006637708 
 |   
 |
|
 PCT\_USABLE\_BASES\_ON\_BAIT 
 | 0.633581 
 | **1.98479096** 
 |
|
 PCT\_USABLE\_BASES\_ON\_TARGET 
 | 0.319218 
 |   
 |
|
 MEAN\_BAIT\_COVERAGE 
 | 65.877482 
 | **1.98503292** 
 |
|
 MEAN\_TARGET\_COVERAGE 
 | 33.187098 
 |   
 |

I set the parameter to --NEAR\_DISTANCE 0 --MINIMUM\_BASE\_QUALITY 0 -MQ 0 but no changes observed.

CMD="gatk $JAVAOPT CollectHsMetrics -I $INBAM -O $PREFIX.HsMetrics -BI **$BEDFILE.interval\_list** -TI **$BEDFILE.interval\_list** --NEAR\_DISTANCE 0 --MINIMUM\_BASE\_QUALITY 0 -MQ 0 --REFERENCE\_SEQUENCE $REF"

I'll appreciate it if you could help understand the difference.

Thank you in advance.

Best,

Ramzi<br><br><i>(created from <a href='https://broadinstitute.zendesk.com/agent/tickets/5181'>Zendesk ticket #5181</a>)<br>gz#5181</i>

**Comment by bhanugandham** [comment]
Hi @yfarjoun I created a issue ticket for this based on GATK office hrs discussion.

**Comment by rtemanni** [comment]
Hello Picard Team,
I was wondering if you got a chance to looks at this issue?
Thank you 
Ramzi

**Comment by yfarjoun** [comment]
Sorry for the delay. Despite the description of the metric the code indicates that 

- The TARGET metrics ignore duplicate reads  (as described)
- The BAIT metrics do not (which runs contrary to the description)

@tfenne @nh13 @ktibbett @lbergelson 

What's your take on this? Should there be a difference regarding duplicate status reads wrt to the following metrics:
* ON_PROBE_BASES<->ON_BAIT_BASES<->ON_TARGET_BASES
* PCT_USABLE_BASES_ON_BAIT<->PCT_USABLE_BASES_ON_TARGET 
* MEAN_TARGET_COVERAGE <->MEAN_BAIT_COVERAGE

Currently, the javaDoc of these metrics either doesn't specify whether it pertains to deduped reads or not, or claims that both the BAIT and the TARGET metrics use deduped reads. in reality, the code uses deduped reads for target metrics and all reads for bait/probe metrics. 

We might want to either fix the description or fix the code...

**Comment by yfarjoun** [comment]
Thanks @rtemanni for bringing this to our attention.

**Comment by lbergelson** [comment]
It seems like they should be consistent.  We probably care more about the de-duplicated values so I would probably make both use that.  At least PCT_USABLE_BASES implies to me that it should be deduplicated.   Is it useful to have separate numbers for each with / without duplicates?

**Comment by yfarjoun** [comment]
I think that bait coverage ignored the duplicate flag on purpose, as it supposed to tell us something about the efficiency of the bait itself. 

My view is that we should fix the JavaDoc....not the code, in this case.....

**Comment by lbergelson** [comment]
It seems confusing even with do changes...  Why not add a second metric for each with/without duplicates?

**Comment by yfarjoun** [comment]
Because then we will have to start fielding questions about which metric to
use...I think that we should have metrics that are useful and well designed
for their purpose.

The reason to look at bait coverage is that you want to know how well your
baits are doing, the reason to look at target coverage is to know if you
have (useful) coverage where you had hoped....so the metrics should be
designed to best answer those questions.

my opinion.... :-)

**Comment by lbergelson** [comment]
Not just a grab-bag of random numbers of questionable utility?

**Comment by tfenne** [comment]
I agree with @yfarjoun that we should fix the documentation (which was at one point, I believe, correct) and leave the code as is.

The general idea here is that when you're talking about _targets_ you are generally concerned about the utility of data in downstream applications such as short-variant or CNV calling.  Therefore you just want to know about how much coverage you have post-dedupe, how that coverage varies etc.  

And that when you are talking about _baits_ you are _generally_ interested in optimizing the sample-preparation process.  It's not 100% clear that using all (vs. unique) reads is always better for this purpose, but given that most hybsel processes include a PCR before the capture (i.e. bait-involved) step, it's not clear that excluding duplicate reads is the right choice.  It has, in the past, been useful to have the bait-oriented stats be based on all reads.  Changing that now would be very confusing and would largely just duplicate the target stats.

**Comment by rtemanni** [comment]
Dear @yfarjoun & @tfenne 
it might be helpful to have an example that explains how the different fields are calculated. That would be very informative
Below an example where BAIT=Target
Thank you once again for your help.
Best,
Ramzi


SampleID | 200298831
-- | --
SampleID_LaneNumber | 200298831
BAIT_SET | HVKWJBBXX
GENOME_SIZE | 3137454505
**BAIT_TERRITORY | 60456963**
**TARGET_TERRITORY | 60456963**
BAIT_DESIGN_EFFICIENCY | 1
TOTAL_READS | 208575876
PF_READS | 208575876
PF_UNIQUE_READS | 148126592
PCT_PF_READS | 1
PCT_PF_UQ_READS | 0.710181
PF_UQ_READS_ALIGNED | 147498028
PCT_PF_UQ_READS_ALIGNED | 0.995757
PF_BASES_ALIGNED | 30024627028
PF_UQ_BASES_ALIGNED | 21374140145
ON_BAIT_BASES | 10550966146
NEAR_BAIT_BASES | 4423951850
OFF_BAIT_BASES | 15049709032
ON_TARGET_BASES | 5738538845
PCT_SELECTED_BASES | 0.498755
PCT_OFF_BAIT | 0.501245
ON_BAIT_VS_SELECTED | 0.704576
MEAN_BAIT_COVERAGE | 174.520281
MEAN_TARGET_COVERAGE | 94.919403
MEDIAN_TARGET_COVERAGE | 89
PCT_USABLE_BASES_ON_BAIT | 0.337238
PCT_USABLE_BASES_ON_TARGET | 0.18342
FOLD_ENRICHMENT | 18.236678
ZERO_CVG_TARGETS_PCT | 0.016119
PCT_EXC_DUPE | 0.288113
PCT_EXC_MAPQ | 0.038605
PCT_EXC_BASEQ | 0.011976
PCT_EXC_OVERLAP | 0.152531
PCT_EXC_OFF_TARGET | 0.350779
FOLD_80_BASE_PENALTY | 1.636541
PCT_TARGET_BASES_1X | 0.982852
PCT_TARGET_BASES_2X | 0.981056
PCT_TARGET_BASES_10X | 0.974659
PCT_TARGET_BASES_20X | 0.964395
PCT_TARGET_BASES_30X | 0.946617
PCT_TARGET_BASES_40X | 0.916002
PCT_TARGET_BASES_50X | 0.865698
PCT_TARGET_BASES_100X | 0.402988
HS_LIBRARY_SIZE | 73998597
HS_PENALTY_10X | 6.375295
HS_PENALTY_20X | 6.685376
HS_PENALTY_30X | 7.055088
HS_PENALTY_40X | 7.472504
HS_PENALTY_50X | 7.98056
HS_PENALTY_100X | 13.934111
AT_DROPOUT | 2.035285
GC_DROPOUT | 1.650626
HET_SNP_SENSITIVITY | 0.965551
HET_SNP_Q | 15

**Comment by alanhoyle** [comment]
I think part of the issue here is that different nomenclature is used by different groups. 

I've linked what I can find for the definitions from various groups:  I think one of the problems is that the Picard definitions are less clear than either Agilent or Illumina's definitions.  

Can someone from Broad confirm that the below is correct? 

Thing | [HsMetrics](https://gatk.broadinstitute.org/hc/en-us/articles/360036856051-CollectHsMetrics-Picard-) | [Agilent](https://earray.chem.agilent.com/suredesign/help/Glossary_of_terms.htm) | [Illumina](https://support.illumina.com/sequencing/sequencing_kits/nextera-rapid-capture-exome-kit/downloads.html) | i.e
------ | ------- | ------ | ----- | -----
Probe | BAIT | Target | Probes | the genomic coordinates of the actual probes used in the capture
Capture | TARGET | Covered | Targeted Regions | the genomic coordinates of the areas around the probes that we expect to be captured

**Comment by yfarjoun** [comment]
This is correct other than I would say that TARGET is "the genomic coordinates of the areas for which the BAIT were designed to generate coverage"

**Comment by alanhoyle** [comment]
Thanks, @yfarjoun : I'll create a new documentation request to clarify this.

**Comment by georgecresswell** [comment]
Hello,

I am hijacking the thread just to ask a question related to these definitions, can `(MEAN_BAIT_COVERAGE - MEAN_TARGET_COVERAGE) / MEAN_BAIT_COVERAGE`, be interpreted as the % of duplicate coverage on the target region, if you use the same bed file for both bait and target? As I understand it PCT_EXC_DUPE is the duplicate % for the whole genome regardless of the target/bait regions.

George

**Comment by murphycj2** [comment]
Another suggestion is to just add a parameter to the tool allows the user to turn off the duplicate filtering when it does its `target` calculations. Thoughts?
